src/Components/Web/src/Virtualization/Virtualize.cs

```diff
@@ -0,0 +1,146 @@
+        {
+            if (firstRender)
+            {
+                _selfReference = DotNetObjectReference.Create<Virtualize<TItem>>(this);
+                await JSRuntime.InvokeVoidAsync($"{JsFunctionsPrefix}.init", _selfReference, _spacerBefore, _spacerAfter);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            var itemsAfter = Math.Max(0, ItemSource.Count() - _itemsVisible - _itemsBefore);
+
+            builder.OpenElement(0, "div");
+            builder.AddAttribute(1, "style", GetSpacerStyle(_itemsBefore));
+            builder.AddElementReferenceCapture(2, elementReference => _spacerBefore = elementReference);
+            builder.CloseElement();
+
+            builder.OpenRegion(3);
+
+            if (ItemTemplate != null)
+            {
+                var items = IsVirtualized ? ItemSource.Skip(_itemsBefore).Take(_itemsVisible) : ItemSource;
+
+                foreach (var item in items)
+                {
+                    ItemTemplate(item)(builder);
+                }
+            }
+
+            builder.CloseRegion();
+
+            builder.OpenElement(4, "div");
+            builder.AddAttribute(5, "style", GetSpacerStyle(itemsAfter));
+            builder.AddElementReferenceCapture(6, elementReference => _spacerAfter = elementReference);
+
+            builder.CloseElement();
+
+            if (Footer != null)
+            {
+                builder.OpenRegion(7);
+
+                Footer.Invoke(builder);
+
+                builder.CloseRegion();
+            }
+        }
+
+        /// <summary>
+        /// Called when the top spacer becomes visible.
+        /// This method is intended to be invoked only from JavaScript.
+        /// </summary>
+        /// <param name="spacerSize">The new top spacer size.</param>
+        /// <param name="containerSize">The top spacer's container size.</param>
+        [JSInvokable]
+        public void OnTopSpacerVisible(float spacerSize, float containerSize)
+        {
+            CalcualteItemDistribution(spacerSize, containerSize, out _itemsVisible, out _itemsBefore);
+
+            StateHasChanged();
+        }
+
+        /// <summary>
+        /// Called when the bottom spacer becomes visible.
+        /// This method is intended to be invoked only from JavaScript.
+        /// </summary>
+        /// <param name="spacerSize">The new bottom spacer size.</param>
+        /// <param name="containerSize">The bottom spacer's container size.</param>
+        [JSInvokable]
+        public void OnBottomSpacerVisible(float spacerSize, float containerSize)
+        {
+            CalcualteItemDistribution(spacerSize, containerSize, out _itemsVisible, out var itemsAfter);
+
+            _itemsBefore = Math.Max(0, ItemSource.Count() - itemsAfter - _itemsVisible);
+
+            if (itemsAfter == 0)
+            {
+                _ = FetchItemsAsync();
```

Comment:
I don't know for sure how you want to do the error handling, but probably here you might want to `await FetchItemsAsync()`.